### PR TITLE
setup docs updates and auto-generation of xcconfig file

### PR DESCRIPTION
**Pull request by jlipps**
@imurchie have a look and let me know what you think

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,19 +86,32 @@ Setting `noReset` to `true` will leave the simulator running at the end of a tes
 The `appium-xcuitest-driver` has provisional support for iOS real devices. Not all functionality is currently supported.
 
 WebDriverAgent needs to be built with `development team` and `provisioning profile` installed on device. The easiest way
-to do this is to specify them in an xcconfig file, the path to which is passed in to the system using the
-`xcodeConfigFile` desired capability
+to do this is to specify them as desired capabilities:
+
+```
+{
+  "xcodeOrgId": "<Your Team ID>",
```
Comment (imurchie) [inline README.md]: I think either this should be "Team ID" or on line 94 it should be "Your Team ID".

File: README.md [hunk 2]
```diff
@@ -85,19 +86,32 @@ Setting `noReset` to `true` will leave the simulator running at the end of a tes
...
+
+In either case, the Team ID is a unique 10-character string generated by Apple that is assigned to your team. You can find your Team ID using your developer account (Sign in to [developer.apple.com/account](developer.apple.com/account), and click Membership in the sidebar. Your Team ID appears in the Membership Information section under the team name. You can also find your team ID listed under the "Organizational Unit" field in your iPhone Developer certificate in your keychain).
+
+Note that these are mutually exclusive strategies; use _either_ the `xcodeConfigFile` capability or the combination of `xcodeOrgId` and `xcodeSigningId`.
 
 ### Manual configuration alternative
 
 Alternatively, the profile can be manually associated with the project (keep in mind that this will have to be done each
-time the WebDriverAgent is updated):
+time the WebDriverAgent is updated, and is _not_ recommended):
```
Comment (imurchie) [inline README.md]: 👍

File: README.md [hunk 3]
```diff
@@ -176,6 +190,8 @@ Differences noted here
 |`showXcodeLog`|Whether to display the output of the Xcode command used to run the tests. If this is `true`, there will be **lots** of extra logging at startup. Defaults to `false`|e.g., `true`|
 |`realDeviceLogger`|Device logger for real devices. It could be path to `deviceconsole` (installed with `npm install deviceconsole`, a compiled binary named `deviceconsole` will be added to `./node_modules/deviceconsole/`) or `idevicesyslog` (comes with libimobiledevice). Defaults to `idevicesyslog`|`idevicesyslog`, `/abs/path/to/deviceconsole`|
 |`iosInstallPause`|Time in milliseconds to pause between installing the application and starting WebDriverAgent on the device. Used particularly for larger applications. Defaults to `0`|e.g., `8000`|
+|`xcodeOrgId`|Apple developer team identifier string. Must be used in conjunction with `xcodeSigningId` to take effect.|e.g., `JWL241K123`|
+|`xcodeSigningId`|String representing a signing certificate. Must be used in conjunction with `xcodeOrgId`. This is usually just "iPhone Developer"|e.g., `iPhone Developer`|
```
Comment (imurchie) [inline README.md]: Should we default it to "iPhone Developer"?

File: lib/utils.js [hunk 4]
```diff
@@ -90,4 +90,18 @@ async function adjustWDAAttachmentsPermissions (perms) {
   }
 }
 
-export { detectUdid, getAndCheckXcodeVersion, getAndCheckIosSdkVersion, killAppUsingAppName, adjustWDAAttachmentsPermissions };
+async function generateXcodeConfigFile (orgId, signingId) {
+  log.debug(`Generating xcode config file for orgId ${orgId} and signingId ` +
```
Comment (imurchie) [inline lib/utils.js]: Can you put the `orgId` and `signingId` in single quotes? It makes it easier to see at a glance if something is wrong with the values (like they accidentally put a space in it or something).

File: lib/utils.js [hunk 5]
```diff
@@ -90,4 +90,18 @@ async function adjustWDAAttachmentsPermissions (perms) {
...
+            `${signingId}`);
+  let contents = `DEVELOPMENT_TEAM = ${orgId}
+CODE_SIGN_IDENTITY = ${signingId}
+`;
+  let xcconfigPath = await tempDir.path('appium-temp.xcconfig');
+  log.debug(`Writing xcode config file to ${xcconfigPath}`);
+  await fs.writeFile(xcconfigPath, contents);
```
Comment (mykola-mokhnach) [inline lib/utils.js]: shall we explicitly provide encoding here?

**Comment by jlipps** [comment]
alright, comments addressed. thanks for jumping in on these reviews @mykola-mokhnach

**Comment by imurchie** [comment]
LGTM.

**Comment by jlipps** [comment]
@imurchie merged but since you're about to merge as well i'll let you publish :-)

**Pull request merged**